src/main/kotlin/org/rust/ide/intentions/import/ImportNameIntention.kt

```diff
@@ -32,31 +33,48 @@ class ImportNameIntention : RsElementBaseIntentionAction<ImportNameIntention.Con
         if (basePath.reference.resolve() != null) return null
         val pathMod = path.containingMod
         val pathSuperMods = HashSet(pathMod.superMods)
-        // TODO: support reexports
-        val candidates = RsNamedElementIndex.findElementsByName(project, basePath.referenceName)
+
+        val explicitItems = RsNamedElementIndex.findElementsByName(project, basePath.referenceName)
             .asSequence()
             .filterIsInstance<RsQualifiedNamedElement>()
-            .filter { basePath != path || !(it is RsMod || it is RsModDeclItem || it.parent is RsMembers) }
-            .mapNotNull { item -> (item as? RsVisible)?.canBeImported(pathSuperMods)?.let { ImportCandidate(item, it) } }
+            .map { ImportItem.ExplicitItem(it) }
+
+        val reexportedItems = RsReexportIndex.findReexportsByName(project, basePath.referenceName)
+            .asSequence()
+            .mapNotNull {
+                val item = it.path?.reference?.resolve() as? RsQualifiedNamedElement ?: return@mapNotNull null
+                ImportItem.ReexportedItem(it, item)
+            }
+
+        val candidates = (explicitItems + reexportedItems)
+            .filter { basePath != path || !(it.item is RsMod || it.item is RsModDeclItem || it.item.parent is RsMembers) }
+            .mapNotNull { importItem -> importItem.canBeImported(pathSuperMods)?.let { ImportCandidate(importItem, it) } }
             // check that result after import can be resolved and resolved element is suitable
             // if no, don't add it in candidate list
             .filter { path.canBeResolvedToSuitableItem(project, pathMod, it.info) }
             .toList()
+
         if (candidates.isEmpty()) return null
         return Context(path, candidates)
     }
 
-    // Semantic signature of method is `RsVisible.canBeImported(mod: RsMod)`
+    // Semantic signature of method is `ImportItem.canBeImported(mod: RsMod)`
     // but in our case `mod` is always same and `mod` needs only to get set of its super mods
     // so we pass `superMods` instead of `mod` for optimization
-    private fun RsVisible.canBeImported(superMods: Set<RsMod>): ImportInfo? {
-        val parentMod = (if (this is RsMod) `super` else containingMod) ?: return null
+    private fun ImportItem.canBeImported(superMods: Set<RsMod>): ImportInfo? {
+        if (item !is RsVisible) return null
+
+        val parentMod = parentMod ?: return null
         val ourSuperMods = parentMod.superMods
 
         // try to find latest common ancestor module of `parentMod` and `mod` in module tree
         // we need to do it because we can use direct child items of any super mod with any visibility
         val lca = ourSuperMods.find { it in superMods }
-        val crateRelativePath = (this as? RsQualifiedNamedElement)?.crateRelativePath?.removePrefix("::") ?: return null
+        val crateRelativePath = crateRelativePath ?: return null
+        val isPublic = when (this) {
```

Comment:
Can be a virtual function of `ImportItem`.

Reply:
Hm, the reason why I didn't it before is check for `RsVisibile` at the beginning of method and next smart cast in `when` expression
So I can move it into separate method of `ImportItem` but in my opinion, it will be not very gracefully
Anyway, it should be refactored to support enum variants which are not `RsVisible` at all and can't be import now at all.

Reply:
Sounds reasonable!